### DISCUSSION TITLE
Swap devices

### DISCUSSION THREAD

**Discussion by WeepingSomnabulist** [WhiteMagic/JoystickGremlin#521] [Q&A]
Hello,

I've got two T1600M joysticks. I've used Joystick Gremlin to bind them into one vjoy. It's been amazing.

However, today I had to disconnect them and move to different USB ports. Now the bindings for my left stick are on the right, and vice-versa. I presume it's because they have the same device ID. I've tried swapping USB ports, inserting each at different times, and it always shows the left and right ones swapped.

I saw there's a Swap Devices tool but I don't know how to use it. I could simply physically swap them, but the right stick is in better condition and so I prefer to aim with it.

Could you help me please?

**Comment by WhiteMagic** [comment]
Moving devices around USB ports should not affect anything, that's the whole point of Windows' unique device identifiers that Gremlin uses to identify devices.

As to the question of how to use the swap devices utility. The purpose it has is to associate all actions associated to one particular device with another one, effectively doing a search & replace of device identifiers. To trigger that you press the button on the right side of the entry. That then waits for you to press a button on the device you want those actions to be assigned to. In your case where you want to swap them around you just need to press the button in the UI and then press an arbitrary button on your physical joystick.
